server/app/mutations/grids/update.rb

```diff
@@ -54,24 +67,31 @@ def validate
 
     def execute
       attributes = {}
-      if self.stats
-        attributes[:stats] = self.stats
-      end
-      if self.trusted_subnets
+
+      if self.node_count # assume this is a full update
+        attributes[:stats] = self.stats || { statsd: nil }
         attributes[:trusted_subnets] = self.trusted_subnets
-      end
-      if self.default_affinity
         attributes[:default_affinity] = self.default_affinity
+      else # assume this is pre 1.2.2 client that didn't send a full update
+        attributes[:stats] = self.stats if self.stats
+        attributes[:default_affinity] = self.default_affinity if self.default_affinity
       end
 
       if self.logs
         if self.logs[:forwarder] == 'none'
```

Comment:
After changing the `forwarder` to be an `optional` field, the `self.logs[:forwarder]` may now be `nil`. As is, this would result in `attributes[:grid_logs_opts] = GridLogsOpts.new(forwarder: nil, opts: nil)`, which is weird.

IMO better to just accept `self.logs[:forwarder].nil? || self.logs[:forwarder] == 'none'` -> `attributes[:grid_logs_opts] = nil`.